File: jaeseuk/kokotlin/src/main/kotlin/programmers/level1/FactorsNumberAdd.kt
package programmers.level1

class FactorsNumberAdd {
    fun solution(left: Int, right: Int): Int {
        var answer: Int = 0

        for (i in left..right) {
            if (Math.sqrt(i.toDouble()) % 1 > 0) {
                answer += i
            } else {
                answer -= i
            }
        }

        return answer
    }
}

File: jaeseuk/pypython/Programmers/level1/factors_number_add.py
"""
math.sqrt 대신 num ** 0.5 가능
"""

import math


def cnt(num):
    cnt = 0

    for i in range(1, int(math.sqrt(num)) + 1):
        q, r = divmod(num, i)

        if r == 0:
            if q != i:
                cnt += 2
            else:
                cnt += 1

    return cnt


def solution(left, right):
    answer = 0

    for i in range(left, right + 1):
        answer += i if cnt(i) % 2 == 0 else -i

    return answer
